pkg/theiactl/commands/check.go

```diff
@@ -72,12 +78,88 @@ $ theiactl policyreco check --id e998433e-accb-4888-9fc8-06563f073e86
 			return err
 		}
 		state := strings.TrimSpace(string(sparkApplication.Status.AppState.State))
+		if state == "RUNNING" {
+			// Check the working progress of running recommendation job
+			// Forward the policy recommendation service port
+			portForwardCmd := exec.Command("kubectl", "port-forward", fmt.Sprintf("service/policy-reco-%s-ui-svc", recoID), "-n", flowVisibilityNS, "4040:4040")
+			if err := portForwardCmd.Start(); err != nil {
+				return fmt.Errorf("fail to forward port for policy recommendation service, %v", err)
+			}
+			defer portForwardCmd.Process.Kill()
+			stateProgress, err := getPolicyRecommendationProgress(recoID)
+			if err != nil {
+				return err
+			}
+			state += stateProgress
+		}
 		fmt.Printf("Status of this policy recommendation job is %s\n", state)
 		return nil
-		// TODO: add implementation of checking work progress through Spark Monitoring Service after port forwarder finished
 	},
 }
 
+func getPolicyRecommendationProgress(id string) (string, error) {
```

Comment:
is the 'id' passed to this function the same as the SparkAppId that would be retrieved later on?
If so, do we actually need to pass it?

Reply:
I notice there are some duplicated comments, just close them.